controllers/topology/util.go

```diff
@@ -39,6 +39,11 @@ func infrastructureMachineTemplateNamePrefix(clusterName, machineDeploymentTopol
 	return fmt.Sprintf("%s-%s-infra-", clusterName, machineDeploymentTopologyName)
 }
 
+// infrastructureMachineTemplateNamePrefix calculates the name prefix for a InfrastructureMachineTemplate.
+func controlPlaneinfrastructureMachineTemplateNamePrefix(clusterName, controlPlaneName string) string {
+	return fmt.Sprintf("%s-%s-infra-", clusterName, controlPlaneName)
+}
```

Comment:
```suggestion
// infrastructureMachineTemplateNamePrefix calculates the name prefix for a InfrastructureMachineTemplate.
func controlPlaneinfrastructureMachineTemplateNamePrefix(clusterName string) string {
	return fmt.Sprintf("%s-controlplane-", clusterName)
}
```

The same function should be used in compute